NineChronicles.Headless.Executable/Commands/GenesisCommand.cs

```diff
@@ -14,125 +11,211 @@
-                genesisConfig.ActivationKeyCount);
+        private void ProcessAdmin(AdminConfig? config, PrivateKey initialMinter, out AdminState adminState)
+        {
+            // FIXME: If the `adminState` is not required inside `MineGenesisBlock`,
+            //        this logic will be much lighter.
+            Console.WriteLine("Processing admin for genesis...");
+            adminState = new AdminState(new Address(), 0);
+            if (config is null)
+            {
+                Log.Information("AdminConfig not provided. Skip admin setting...");
+                return;
+            }
 
-            if (string.IsNullOrEmpty(genesisConfig.PrivateKey))
+            if (config.Value.Activate)
+            {
+                if (string.IsNullOrEmpty(config.Value.Address))
+                {
+                    Log.Information("Admin address not provided. Give admin privilege to initialMinter");
+                    adminState = new AdminState(initialMinter.ToAddress(), config.Value.ValidUntil);
+                }
+            }
+            else
             {
-                throw Utils.Error("PrivateKey is not set");
+                Log.Information("Inactivate Admin. Skip admin setting...");
             }
 
-            if (string.IsNullOrEmpty(genesisConfig.GoldDistributionPath))
+            Log.Information("Admin config done");
+        }
+
+        private void ProcessExtra(ExtraConfig? config,
+            out List<PendingActivationState> pendingActivationStates
+        )
+        {
+            Console.WriteLine("Processing extra data for genesis...");
+            pendingActivationStates = new List<PendingActivationState>();
+
+            if (config is null)
             {
-                throw Utils.Error("GoldDistributionPath is not set");
+                Log.Information("Extra config not provided");
+                return;
             }
 
-            if (string.IsNullOrEmpty(genesisConfig.AdminAddress))
+            if (!string.IsNullOrEmpty(config.Value.PendingActivationStatePath))
             {
-                throw Utils.Error($"{nameof(genesisConfig.AdminAddress)} is not set");
+                string hex = File.ReadAllText(config.Value.PendingActivationStatePath).Trim();
+                List decoded = (List)_codec.Decode(ByteUtil.ParseHex(hex));
+                CreatePendingActivations action = new();
+                action.LoadPlainValue(decoded[1]);
+                pendingActivationStates = action.PendingActivations.Select(
+                    pa => new PendingActivationState(pa.Nonce, new PublicKey(pa.PublicKey))
+                ).ToList();
             }
+        }
+
+        [Command(Description = "Mine a new genesis block")]
+        public void Mine(
+            [Argument("CONFIG", Description = "JSON config path to mine genesis block")]
+            string configPath)
+        {
+            var loggerConf = new LoggerConfiguration();
```

Comment:
If you intended to write to the stdout, you should build the logger configuration with `WriteTo.Console()` 

```suggestion
            var loggerConf = new LoggerConfiguration()
                .WriteTo.Console();
```

Or I recommend using the `IConsole`-typed `_console` variable. It makes it testable the command.